src/main/java/com/github/polimi_mt_acg/back2school/api/v1/students/StudentsResource.java

```diff
@@ -226,26 +216,24 @@ public Response postStudentGrades(
       return Response.status(Status.NOT_FOUND).build();
     }
 
-    // Fetch User
-    Optional<User> studentOpt = fetchStudent(Integer.parseInt(studentId), session);
-    if (!studentOpt.isPresent()) {
+    // Fetch student
+    User student = session.get(User.class, Integer.parseInt(studentId));
+    if (student == null) {
```

Comment:
I'll comment on this here but it holds for the rest of the code. I prefer `Optional<T>` better than having a method the returns a possibly `null` reference. Why so? `Optional<T>` embeds the notion of _the object may or may not be there_ forcing the programmer to structure the code in a way that always checks if it contains a valid value or not.
You may say: "Yes, but I'm checking whether the return value is `null`" and that's the one more argument of the programmer that is about to get a `NullPointerException` right in his face. 

**Long story short**:

Have `DatabaseHandler.fetchEntityBy()` wrapping a call to `session.get()` and return and Optional<T> which contains the element if the result != null, or empty otherwise.

Reply:
I got your point and actually I have to admit that I agree with you... that's the bad of Java unfortunately. 
The reason I was suggesting this edit was because of simplicity and less verbosity of the code. Sorry, I'm used to write less and I would like to apply the same way of doing things to Java.. but I understand this is not possible.

Reverted at: https://github.com/polimi-mt-acg/back2school/commit/066791eba43e06ab2c13cbb59c2fc758ee927839